code/modules/paperwork/stamps.dm

```diff
@@ -1,55 +1,46 @@
 /obj/item/stamp
 	name        = "rubber stamp"
 	desc        = "A rubber stamp for stamping important documents."
-	icon        = 'icons/obj/items/rubber_stamps.dmi'
-	icon_state  = "stamp-deckchief"
-	item_state  = "stamp"
-	throwforce  = 0
+	icon        = 'icons/obj/items/stamps/stamp_deckchief.dmi'
+	icon_state  = ICON_STATE_WORLD
+	throwforce  = 1
 	w_class     = ITEM_SIZE_TINY
 	throw_speed = 7
 	throw_range = 15
 	material    = /decl/material/solid/metal/steel
+	item_flags  = ITEM_FLAG_STAMP
 	matter      = list(
 		/decl/material/solid/organic/plastic = MATTER_AMOUNT_REINFORCEMENT,
 	)
 	attack_verb = list("stamped")
 
 /obj/item/stamp/denied
 	name = "\improper DENIED rubber stamp"
-	icon_state = "stamp-deny"
-	attack_verb = list("stamped", "denied")
+	icon = 'icons/obj/items/stamps/stamp_deny.dmi'
 
 /obj/item/stamp/clown
 	name = "clown's rubber stamp"
-	icon_state = "stamp-clown"
+	icon = 'icons/obj/items/stamps/stamp_clown.dmi'
 
 /obj/item/stamp/boss
 	name = "boss' rubber stamp"
-	icon_state = "stamp-boss"
+	icon = 'icons/obj/items/stamps/stamp_boss.dmi'
 
 /obj/item/stamp/boss/Initialize()
 	name = "[global.using_map.boss_name]'s' rubber stamp"
 	. = ..()
 
 // Syndicate stamp to forge documents.
 /obj/item/stamp/chameleon/attack_self(mob/user)
-
-	var/list/stamp_types = typesof(/obj/item/stamp) - src.type // Get all stamp types except our own
 	var/list/stamps = list()
-
 	// Generate them into a list
-	for(var/stamp_type in stamp_types)
-		var/obj/item/stamp/S = new stamp_type
-		stamps[capitalize(S.name)] = S
-
+	for(var/stamp_type in subtypesof(/obj/item/stamp))
+		var/obj/item/stamp/S = stamp_type
+		if(!TYPE_IS_ABSTRACT(S))
+			stamps[capitalize(initial(S.name))] = S
 	var/list/show_stamps = list("EXIT" = null) + sortTim(stamps, /proc/cmp_text_asc) // the list that will be shown to the user to pick from
-
 	var/input_stamp = input(user, "Choose a stamp to disguise as.", "Choose a stamp.") in show_stamps
-
-	if(user && (src in user.contents))
-
+	if(input_stamp && !QDELETED(user) && !QDELETED(src) && user.get_active_held_item() == src)
 		var/obj/item/stamp/chosen_stamp = stamps[capitalize(input_stamp)]
-
 		if(chosen_stamp)
-			SetName(chosen_stamp.name)
-			icon_state = chosen_stamp.icon_state
+			appearance = chosen_stamp
```

Comment:
this still didn't work before, but i think the boss stamp won't work properly here. it'll say `boss's rubber stamp` instead of `[global.using_map.boss_name]'s' rubber stamp`.

Reply:
can we... use the atom info repository for this?

Reply:
yep. just do the appearance set and then `SetName(atom_info_repository.get_name_for(chosen_stamp))` probably